### ISSUE TITLE
Break down the "average for the period" categories of the WLM by project code

### ISSUE THREAD

**Issue by Scottan** [UoMResearchIT/CapX#503]
It would be helpful to see a breakdown of how much time on average is spent on each project over a period of time. The same could be useful for the other categories, e.g. to understand how much of the BAU time is spent on training vs. community work or social time.

To avoid clutter, I think this would be best done as some sort of optional "advanced" view. For example, have a button next to each category that creates a drop-down table showing each innate code and average FTE that contributes to the total average for that category.

**Comment by aharwood2** [comment]
I think we can probably do what you need here with a popup when you click an average in the table?

**Comment by Scottan** [comment]
Yeah, that could also work, I'm happy with whatever's easiest to implement. It would be nice to be able to copy-paste it though, so maybe not a tooltip

**Comment by aharwood2** [comment]
I'm thinking now that perhaps a user clicks on a bar on the graph and then a popup appears with a summary of the timesheet tasks for that week and that category with the notes field at the bottom? I guess that should be able to capture the information needed i.e. you are going to sit down with staff and look at the weeks where the bars are large and dissect what happened that week for example.

**Comment by Scottan** [comment]
What you say above sounds useful, but I think it would also be useful to be able to amalgamate over a period more than just a week - I thought if you can break down the Averages for the Period at the bottom by task when you put your mouse over them, by setting a date range you can see how much time was put on any task over any period.

I'm just in a meeting now with @fherreazcue where it would have been really helpful to have been able to do this ;)

**Comment by rowleya** [comment]
> What you say above sounds useful, but I think it would also be useful to be able to amalgamate over a period more than just a week - I thought if you can break down the Averages for the Period at the bottom by task when you put your mouse over them, by setting a date range you can see how much time was put on any task over any period.

It sounds like the view in the timesheets might actually be nicer to look at than the one in the WLM analysis, where longer periods probably make more sense.

**Comment by aharwood2** [comment]
Based on the discussion in the linked issue, I think I'm going to "downgrade" this issue to the following:
- [x] Add click on bar navigation to the timesheet
- [ ] Add summary on the table as to what the most popular task was that was booked against in each category (maybe with a dialog when clicking on a value)
- [ ] Add a hint on the table to outline the above feature
- [ ] Add an API endpoint that allows timesheet data e.g. `GetTimesheetEntriesForPersonForDateRange()` to be pulled for a user (assuming you have access to their data) for a date range so that more complex visualisations can be created if others want them

**Comment by DrAtbas** [comment]
Yesterday, I pushed in some updates related to this.

When you have time @aharwood2, I would like to discuss my approach! And maybe we can do it so that I can debug this later on 😆

Do you have any time to discuss it on Tuesday?

**Comment by aharwood2** [comment]
Thanks, @DrAtbas. Yes, happy to take a quick look with you if you want to drop something in my calendar. You could always raise a PR for it and I can review it that way too if you want?

**Comment by DrAtbas** [comment]
I haven't tested it yet, so I would like to ensure it works first. Then, we can raise the PR! So I also saw that you auto deployment so I dont want to merge it and break everything 😆

**Comment by aharwood2** [comment]
> I haven't tested it yet, so I would like to ensure it works first. Then, we can raise the PR! So I also saw that you auto deployment so I dont want to merge it and break everything 😆

Raising a PR is safe don't worry. Feature branches are only deployed to the test server when the PR is merged and it won't be merged without a review. The GitHub interface is just easy to provide feedback through is all. You could also raise a draft PR if you think it is not ready, that will also help me figure out what has changed. Could you do that?

**Comment by DrAtbas** [comment]
Raised!

**Comment by aharwood2** [comment]
PR has been approved for this which adds API endpoints to allow timesheets for a date range and a person to be pulled through the API. I know this doesn't add the required UI feature but I think a lot of UI feature requests are probably going to end up as DIY API endpoints to be honest.

**Comment by Scottan** [comment]
> I know this doesn't add the required UI feature but I think a lot of UI feature requests are probably going to end up as DIY API endpoints to be honest.

Honestly that's fine, at this point I think it's more important to improve the underlying structure and performance of CapX than extra bells and whistles.

**Comment by DrAtbas** [comment]
Thank you for fixing the last issues! Just merged the PR.
